src/coreclr/jit/targetwasm.h

```diff
@@ -0,0 +1,284 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+#pragma once
+
+#if !defined(TARGET_WASM)
+#error The file should not be included for this platform.
+#endif
+
+// clang-format off
+#define CPU_LOAD_STORE_ARCH      1
+#define CPU_HAS_FP_SUPPORT       1
+#define CPU_HAS_BYTE_REGS        0
+
+// Currently we don't pass any arguments on the linear memory stack so FEATURE_FIXED_OUT_ARGS is not needed.
+#define FEATURE_FIXED_OUT_ARGS   0       // Preallocate the outgoing arg area in the prolog
+#define FEATURE_STRUCTPROMOTE    1       // JIT Optimization to promote fields of structs into registers
+#define FEATURE_MULTIREG_STRUCT_PROMOTE 1  // True when we want to promote fields of a multireg struct into registers
+#define FEATURE_FASTTAILCALL     0       // Tail calls made as epilog+jmp
+#define FEATURE_TAILCALL_OPT     0       // opportunistic Tail calls (i.e. without ".tail" prefix) made as fast tail calls.
+#define FEATURE_IMPLICIT_BYREFS       1  // Support for struct parameters passed via pointers to shadow copies
+#define FEATURE_MULTIREG_ARGS_OR_RET  0  // Support for passing and/or returning single values in more than one register
+#define FEATURE_MULTIREG_ARGS         0  // Support for passing a single argument in more than one register
+#define FEATURE_MULTIREG_RET          0  // Support for returning a single value in more than one register
+#define MAX_PASS_SINGLEREG_BYTES      8  // Maximum size of a struct passed in a single register (long/double).
+#define MAX_PASS_MULTIREG_BYTES       0  // Maximum size of a struct that could be passed in more than one register
+#define MAX_RET_MULTIREG_BYTES        0  // Maximum size of a struct that could be returned in more than one register (Max is an HFA or 2 doubles)
+#define MAX_ARG_REG_COUNT             1  // Maximum registers used to pass a single argument in multiple registers.
+#define MAX_RET_REG_COUNT             1  // Maximum registers used to return a value.
+#define MAX_MULTIREG_COUNT            2  // Maximum number of registers defined by a single instruction (including calls).
+                                         // This is also the maximum number of registers for a MultiReg node.
+
+#define NOGC_WRITE_BARRIERS      0       // No specialized WriteBarrier JIT Helpers
+#define USER_ARGS_COME_LAST      1
+#ifdef TARGET_WASM32
+#define TARGET_POINTER_SIZE      4       // equal to sizeof(void*) and the managed pointer size in bytes for this target
+#else
+#define TARGET_POINTER_SIZE      8       // equal to sizeof(void*) and the managed pointer size in bytes for this target
+#endif
+#define ETW_EBP_FRAMED           0       // No frame pointer chaining on WASM
+
+// TODO-WASM-CQ: measure if "CSE_CONSTS" is beneficial.
+#define CSE_CONSTS               1       // Enable if we want to CSE constants
+#define EMIT_TRACK_STACK_DEPTH   1       // TODO-WASM: set to 0.
+#define EMIT_GENERATE_GCINFO     0       // We don't (can't) use code metadata-based GCInfo on WASM
```

Comment:
Q: when should I use `EMIT_GENERATE_GCINFO` instead of `TARGET_WASM`?
A: use it when the code you are conditioning away relates to:
1) Tracking GCness of variables and registers in codegen.
2) Tracking GCness of variables and registers in emit.
3) Emitting GCInfo via GcInfoEncoder.

A useful mental test here is the hypothetical target of "CoreCLR interpreter (code)", which would have `HAS_FIXED_REGISTER_SET` set to `0`, but `EMIT_GENERATE_GCINFO` - `1` (it would need to use variable-size bitsets, something we don't need to deal with for WASM).